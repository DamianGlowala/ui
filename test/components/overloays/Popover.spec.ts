import { describe, it, expect } from 'vitest'
import { UPopover } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Popover', () => {
  it.each([
    ['basic case', {}],
    ['with mode', { mode: 'click' }],
    ['with open', { open: true }],
    ['with disabled', { disabled: true }],
    ['with openDelay', { openDelay: 100 }],
    ['with closeDelay', { closeDelay: 200 }],
    ['with overlay', { overlay: true }],
    ['with popper', { popper: { key: { id: '1' }, locked: false, arrow: true } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Popover.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UPopover)
    expect(html).toMatchSnapshot()
  })
})
