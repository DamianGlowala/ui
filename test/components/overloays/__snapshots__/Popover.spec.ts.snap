// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover > renders basic case correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_0" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with closeDelay correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_10" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with disabled correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_6" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with mode correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_2" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with open correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_4" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with openDelay correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_8" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with overlay correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_12" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;

exports[`Popover > renders with popper correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div class="" role="button" id="headlessui-popover-button-Ht54ZG1A39_14" aria-expanded="false" data-headlessui-state="">label</div>
  <!---->
  <!--v-if-->
  <!--v-if-->
</div>
<div style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;"></div>"
`;
